homeassistant/components/hydrawise/binary_sensor.py

```diff
@@ -67,3 +86,6 @@ def _update_attrs(self) -> None:
         elif self.entity_description.key == "is_watering":
             zone: Zone = self.zone
             self._attr_is_on = zone.scheduled_runs.current_run is not None
+        elif self.entity_description.key == "rain_sensor":
+            sensor: Sensor = self.sensor
+            self._attr_is_on = sensor.status.active
```

Comment:
I am not sure how many entities you're going to add, but a common pattern is to extend the entity description to provide a value_fn. This will then make sure all logic is written at the same place as the rest of the sensor definition.

Reply:
Ok. Not sure I'm a big fan to add all these lambdas. Gets pretty unwieldy too if the function is not trivial. However, I did rewrite the function to follow the pattern we already implemented for sensors.